phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexToolTimeRangeIT.java

```diff
@@ -0,0 +1,215 @@
+            conn.commit();
+            incrementEdgeByOne();
+            conn.createStatement().execute(String.format(CREATE_INDEX_DDL, indexTableName,
+                    dataTableFullName));
+            conn.commit();
+        }
+    }
+
+    private static void incrementEdgeByOne() {
+        customEdge.incrementValue(1);
+        EnvironmentEdgeManager.injectEdge(customEdge);
+    }
+
+    private static void populateDataTable() throws SQLException {
+        try (Connection conn = getConnection()) {
+            //row 1-> time 4, row 2-> time 5, row 3-> time 6, row 4-> time 7, row 5-> time 8
+            for (int i=0; i<5; i++) {
+                incrementEdgeByOne();
+                PreparedStatement ps = conn.prepareStatement(
+                        String.format(UPSERT_TABLE_DML, dataTableFullName));
+                ps.setInt(1, i+1);
+                ps.setInt(2,(i+1)*10);
+                ps.setInt(3, (i+1)*100);
+                ps.execute();
+                conn.commit();
+            }
+        }
+    }
+
+    private static void setupMiniCluster() throws Exception {
+        Map<String, String> serverProps = Maps.newHashMapWithExpectedSize(2);
+        serverProps.put(QueryServices.STATS_GUIDEPOST_WIDTH_BYTES_ATTRIB, Long.toString(20));
+        serverProps.put(QueryServices.MAX_SERVER_METADATA_CACHE_TIME_TO_LIVE_MS_ATTRIB,
+                Long.toString(5));
+        serverProps.put(QueryServices.EXTRA_JDBC_ARGUMENTS_ATTRIB,
+                QueryServicesOptions.DEFAULT_EXTRA_JDBC_ARGUMENTS);
+        serverProps.put(QueryServices.INDEX_REBUILD_PAGE_SIZE_IN_ROWS, Long.toString(8));
+        Map<String, String> clientProps = Maps.newHashMapWithExpectedSize(2);
+        clientProps.put(QueryServices.USE_STATS_FOR_PARALLELIZATION, Boolean.toString(true));
+        clientProps.put(QueryServices.STATS_UPDATE_FREQ_MS_ATTRIB, Long.toString(5));
+        clientProps.put(QueryServices.TRANSACTIONS_ENABLED, Boolean.TRUE.toString());
+        clientProps.put(QueryServices.FORCE_ROW_KEY_ORDER_ATTRIB, Boolean.TRUE.toString());
+        setUpTestDriver(new ReadOnlyProps(serverProps.entrySet().iterator()),
+                new ReadOnlyProps(clientProps.entrySet().iterator()));
+    }
+
+    private int countRowsInIndex() throws SQLException {
+        incrementEdgeByOne();
+        String select = "SELECT COUNT(*) FROM "+indexTableFullName;
+        try(Connection conn = getConnection()) {
+            ResultSet rs = conn.createStatement().executeQuery(select);
+            while(rs.next()) {
+                return rs.getInt(1);
+            }
+        }
+        return -1;
+    }
+
+    private static Connection getConnection() throws SQLException {
+        return DriverManager.getConnection(getUrl(),
+                PropertiesUtil.deepCopy(TEST_PROPERTIES));
+    }
+
+    @Before
+    public void beforeTest() {
+        incrementEdgeByOne();
+    }
+
+    @Test
+    public void testValidTimeRange() throws Exception {
```

Comment:
Nice set of tests @swaroopak! Is there an error-case test where starttime > endtime?

Reply:
Thanks! :) Yes, I have added them to the IndexToolTest class.